CHANGELOG.md

```diff
@@ -16,6 +16,7 @@
   * `cortex_ruler_clients`
   * `cortex_ruler_client_request_duration_seconds`
 * [ENHANCEMENT] Query-frontend/scheduler: added querier forget delay (`-query-frontend.querier-forget-delay` and `-query-scheduler.querier-forget-delay`) to mitigate the blast radius in the event queriers crash because of a repeatedly sent "query of death" when shuffle-sharding is enabled. #3901
+* [ENHANCEMENT] Querier / ruler: optimized PromQL query engine. #3934 #3989
```

Comment:
Maybe "some optimizations to PromQL query engine"?
It would be better if #3934 listed what those were.

Reply:
What's about now?